modules/EvseV2G/connection.cpp

```diff
@@ -985,3 +1021,51 @@ int connection_start_servers(struct v2g_context* ctx) {
 
     return 0;
 }
+int create_udp_socket(const int udp_port) {
+    // find suitable addresses
+    addrinfo hints;
+    addrinfo *result, *result_pointer;
+    int udp_socket;
+
+    memset(&hints, 0, sizeof(addrinfo));
+    hints.ai_family = AF_UNSPEC;    // allow IPv4 or IPv6
+    hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+    hints.ai_flags = AI_ALL;        // For IPv4 or IPv6
+    hints.ai_protocol = 0;          // OS selects best protocol (best practise)
+
+    // convert port to string
+    char port_str[6];
+    snprintf(port_str, sizeof(port_str), "%d", udp_port);
+
+    //              alternatively:
+    //              "localhost"
+    auto get_addrinfo_result = getaddrinfo(nullptr, port_str, &hints, &result);
+
+    if (get_addrinfo_result != 0) {
+        dlog(DLOG_LEVEL_ERROR, "getaddrinfo: %s", gai_strerror(get_addrinfo_result));
+        return -1;
+    }
+
+    // create the socket
+    for (result_pointer = result; result_pointer != nullptr; result_pointer = result_pointer->ai_next) {
+        udp_socket = socket(result_pointer->ai_family, result_pointer->ai_socktype, result_pointer->ai_protocol);
+
+        if (udp_socket == -1) {
+            continue;
+        }
+
+        if (connect(udp_socket, result_pointer->ai_addr, result_pointer->ai_addrlen) == 0) {
+            break;
+        }
+
+        close(udp_socket);
+    }
+
+    if (result_pointer == nullptr) {
+        dlog(DLOG_LEVEL_ERROR, "Could not connect");
+        return -1;
+    }
+
+    freeaddrinfo(result);
```

Comment:
probably a memory leak, because you might return earlier

Reply:
I think it wouldn't be a memory leak as in any other return case the `result` should be a nullptr

Reply:
Ah, well then it is even more wrong :).  `freeaddrinfo` needs to be called on the linked list, returned by `getaddrinfo`, so the initial one.  Calling `freeaddrinfo` on members of the linked list is probably undefined behaviour (UB).